src/AddImageChecks.cpp

```diff
@@ -211,7 +211,8 @@ Stmt add_image_checks(Stmt s,
             vector<string> extern_users;
             for (size_t i = 0; i < order.size(); i++) {
                 Function f = env.find(order[i])->second;
-                if (f.has_extern_definition()) {
+                if (f.has_extern_definition() &&
```

Comment:
This combination is showing up in a number of places in this PR and it may be more readable to introduce a single well named method to return what this is testing for.